### PR TITLE
feat(ui): Add select all to teams

### PR DESCRIPTION
**Pull request by maheskett**
Add select all for teams when adding a member to an organization or editing member details

screen shot 2018-10-26 at 1 17 22 pm

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/index.jsx [hunk 1]
```diff
@@ -203,6 +203,26 @@ const InviteMember = createReactClass({
     });
   },
 
+  handleSelectAll() {
+    let {teams} = this.getOrganization();
+    let {selectedTeams} = this.state;
+    let allSelected = teams.length === selectedTeams.size;
+
+    if (allSelected) {
+      teams.map(team => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/index.jsx]: Can we just clear the set?
```suggestion
      selectedTeams.clear()
```

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/index.jsx [hunk 2]
```diff
@@ -203,6 +203,26 @@ const InviteMember = createReactClass({
...
+        if (selectedTeams.has(team.slug)) {
+          selectedTeams.delete(team.slug);
+        }
+      });
+    } else {
+      teams.map(team => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/index.jsx]: likewise, can we just create a new set here from teams?
```suggestion
      selectedTeams = new Set(teams.map(({slug}) => slug));
```

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx [hunk 3]
```diff
@@ -3,31 +3,40 @@ import React from 'react';
...
       <Panel className="new-invite-team">
-        <PanelHeader>{t('Team')}</PanelHeader>
+        <PanelHeader hasButtons>
+          {t('Team')}
+          {onSelectAll && (
+            <Button
+              className="select-all"
+              size="small"
+              disabled={disabled}
+              onClick={() => onSelectAll()}
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx]: I don't think the closure here is necessary

```suggestion
              onClick={onSelectAll}
```

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx [hunk 4]
```diff
@@ -3,31 +3,40 @@ import React from 'react';
...
     if (teams.length < 2) return null;
 
     return (
       <Panel className="new-invite-team">
-        <PanelHeader>{t('Team')}</PanelHeader>
+        <PanelHeader hasButtons>
+          {t('Team')}
+          {onSelectAll && (
+            <Button
+              className="select-all"
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx]: Let's use `data-test-id` attribute for testing

```suggestion
              data-test-id="select-all"
```

File: src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx [hunk 5]
```diff
@@ -3,31 +3,40 @@ import React from 'react';
...
+        <PanelHeader hasButtons>
+          {t('Team')}
+          {onSelectAll && (
+            <Button
+              data-test-id="select-all"
+              size="small"
+              disabled={disabled}
+              onClick={onSelectAll}
+            >
+              <SelectAll>{t('Select All')}</SelectAll>
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/organizationMembers/inviteMember/teamSelect.jsx]: Should this text change when all teams are selected?

**Review by billyvg — APPROVED** [review]
lgtm, thanks!

**Pull request merged**